File: leetcode/01.two sum/Solution.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = HashMap<Int, Int>()
        for (i in 0..(nums.size - 1)) {
            if (map.containsKey(nums[i])) {
                return intArrayOf(map[nums[i]]!!, i)
            }
            map[target - nums[i]] = i
        }
        return intArrayOf(-1, -1)
    }
}

File: leetcode/01.two sum/solution.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mapping = {}
        for i, value in enumerate(nums):
            diff = target - nums[i]
            if diff in mapping:
                return [mapping[diff], i]
            else:
                mapping[value] = i
# @lc code=end